ArduCopter/mode_autorotate.cpp

```diff
@@ -111,145 +164,194 @@ void ModeAutorotate::run()
+            g2.arot.set_col_cutoff_freq(g2.arot.get_col_entry_freq());
 
-                // Target head speed is set to rpm at initiation to prevent unwanted changes in attitude
-                _target_head_speed = _initial_rpm/g2.arot.get_hs_set_point();
+            // Target head speed is set to rpm at initiation to prevent unwanted changes in attitude
+            _target_head_speed = _initial_rpm/g2.arot.get_hs_set_point();
 
-                // Set desired forward speed target
-                g2.arot.set_desired_fwd_speed();
+            // Set desired forward speed target
+            g2.arot.set_desired_fwd_speed();
 
-                // Prevent running the initial entry functions again
-                _flags.entry_initial = false;
+            // Prevent running the initial entry functions again
+            _flags.entry_initial = false;
 
-            }
+        }
 
+        if (!hover_autorotation) {
             // Slowly change the target head speed until the target head speed matches the parameter defined value
-            if (g2.arot.get_rpm() > HEAD_SPEED_TARGET_RATIO*1.005f  ||  g2.arot.get_rpm() < HEAD_SPEED_TARGET_RATIO*0.995f) {
-                _target_head_speed -= _hs_decay*G_Dt;
+            float norm_rpm = g2.arot.get_rpm()/g2.arot.get_hs_set_point();
+            if (norm_rpm > HEAD_SPEED_TARGET_RATIO*1.005f  ||  norm_rpm < HEAD_SPEED_TARGET_RATIO*0.995f) {
+                _target_head_speed -= _hs_decay * last_loop_time_s;
             } else {
                 _target_head_speed = HEAD_SPEED_TARGET_RATIO;
             }
-
             // Set target head speed in head speed controller
             g2.arot.set_target_head_speed(_target_head_speed);
-
             // Run airspeed/attitude controller
-            g2.arot.set_dt(G_Dt);
             g2.arot.update_forward_speed_controller();
-
             // Retrieve pitch target
             _pitch_target = g2.arot.get_pitch();
-
             // Update controllers
-            _flags.bad_rpm = g2.arot.update_hs_glide_controller(G_Dt); //run head speed/ collective controller
+            _flags.bad_rpm = g2.arot.update_hs_glide_controller(); //run head speed/ collective controller
+        } else {
+            _pitch_target = 0.0f;
+            g2.arot.update_hover_autorotation_controller(); //run head speed/ collective controller
+            g2.arot.set_collective_minimum_drag(motors->get_coll_mid());
+            g2.arot.set_entry_sink_rate(curr_vel_z);
+            g2.arot.set_entry_alt(g2.arot.get_ground_distance());
+        }
+
+        break;
+    }
+
+    case Autorotation_Phase::SS_GLIDE: {
+        // Steady state glide functions to be run only once
+        if (_flags.ss_glide_initial == true) {
+
+            gcs().send_text(MAV_SEVERITY_INFO, "SS Glide Phase");
 
-            break;
+            // Set following trim low pass cut off frequency
+            g2.arot.set_col_cutoff_freq(g2.arot.get_col_glide_freq());
```

Comment:
There are a few of these, calling into the lib with a value from the lib. You could either make it one call something like `arot.set_col_cutoff_freq_glide()` or maybe the lib already has enough state to know which value it should use, so you could just do a `update_col_cutoff_freq()`, or you could pass a enum add that state like the `set_target_head_speed` method.